src/components/mod.rs

```diff
@@ -44,3 +46,90 @@ pub use time::Time;
 pub use year_month::YearMonth;
 #[doc(inline)]
 pub use zoneddatetime::ZonedDateTime;
+
+use crate::TemporalError;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u8)]
+pub enum MonthCode {
+    One = 1,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven,
+    Eight,
+    Nine,
+    Ten,
+    Eleven,
+    Twelve,
+    Thirteen,
+}
+
+impl MonthCode {
+    pub fn as_str(&self) -> &str {
```

Comment:
Will these not change in different calendars? Aren't there more month codes in the chinese calendar? such as https://tc39.es/proposal-intl-era-monthcode/

Reply:
Yeah, I realized that earlier. I was sort of thinking we could just expand the enum, but that's probably not sufficient either.

Reply:
Maybe this structure is fine for now and we can refactor it to support more in future? What do you think?

Reply:
I think if you don't have a "leap" flag in the month code then it probably doesn't make sense to have M13... AFAIK no calendars have an M13, if there is a thirteenth month it's always one of M00L through M12L (or maybe M01L through M11L; it's unclear whether M00L and M12L actually exist)